cicd/Jenkins/Jenkinsfile

```diff
@@ -1246,3 +1280,243 @@ def runAdoptionStats() {
+      git clone https://\${USERNAME}:\${ACCESS_TOKEN}@github.com/CenturyLink/po-enterprise-shell.git
+    """, label: "Cloning shell", returnStdout: true)
+    updateBranchChi(version);
+    return version
+  }
+}
+
+def deleteShellFolder() {
+  sh(script: """
+  rm -rf po-enterprise-shell
+  """)
+}
+
+def pushBranchChiVue(version) {
+  withCredentials([[$class: 'UsernamePasswordMultiBinding', credentialsId: env.GITHUB_TOKEN_CREDENTIALS,
+                   usernameVariable: 'USERNAME', passwordVariable: 'ACCESS_TOKEN']]) {
+    sh(script: """
+        set +x
+        git config credential.helper  "!f() {echo username=${env.USERNAME}; echo password=${env.ACCESS_TOKEN}; }"
+        git config --global user.email "cicdint0@lumen.com"
+        cd po-enterprise-shell
+        git status
+        git add package-lock.json package.json
+        git commit -m "New version on ${env.PROJECT_NAME}: ${version}"
+        git push https://\${USERNAME}:\${ACCESS_TOKEN}@github.com/CenturyLink/po-enterprise-shell.git
+        """, returnStdout: true, label: 'pushBranchChiVue')
+  }
+}
+
+def pushBranchChi(version) {
+  withCredentials([[$class: 'UsernamePasswordMultiBinding', credentialsId: env.GITHUB_TOKEN_CREDENTIALS,
+                    usernameVariable: 'USERNAME', passwordVariable: 'ACCESS_TOKEN']]) {
+  sh(script: """
+    set +x
+    git config credential.helper "!f() {echo username=${env.USERNAME}; echo password=${env.ACCESS_TOKEN}; }"
+    git config --global user.email "cicdint0@lumen.com"
+    cd po-enterprise-shell
+    git status
+    git add .env
+    git commit -m "New Chi version: ${version}"
+    git push https://\${USERNAME}:\${ACCESS_TOKEN}@github.com/CenturyLink/po-enterprise-shell.git
+  """, returnStdout: true)
+  }
+}
+
+def updateChiVueOnShell() {
+  def shellJson = readJSON file: 'po-enterprise-shell/package.json', returnPojo: true;
+  def projectJson = readJSON file: 'src/chi-vue/package.json', returnPojo: true;
+  def version = projectJson["version"];
+  if (version != null && version != "" && shellJson["dependencies"]["@centurylink/chi-vue"] != null) {
+    shellJson["dependencies"]["@centurylink/chi-vue"] = version
+    writeJSON file: 'po-enterprise-shell/package2.json', json: shellJson
+    
+    sh(script: """
+    export CHANGED_VERSION=${version}
+    sh run_parser.sh
+    """, returnStdout: true)
+    sh(script: """
+    ls -l
+    ls -l po-enterprise-shell
+    rm po-enterprise-shell/package.json po-enterprise-shell/package2.json
+    mv po-enterprise-shell/package3.json po-enterprise-shell/package.json
+    """, returnStdout: true)
+    return true
+    
+    
+  }
+  return false
+}
+
+def updateChiOnShell() {
+  def projectJson = readJSON file: 'package.json', returnPojo: true;
+  def version = projectJson["version"]
+  sh(script: """
+  ls -l
+  echo "${version}"
+  awk -F= -v version=${version} -f parser_env_shell.awk po-enterprise-shell/.env > po-enterprise-shell/.env2
```

Comment:
please use higher level constructs for these kind of substitutions for easier maintenance

Reply:
I've just added some alternative files, using Python instead of awk. Let's agree on the new setup first and then I'll delete the old ones, so we can easily go back to the old one if necessary